server/services/teamsSummaryCountService.ts

```diff
@@ -0,0 +1,158 @@
+    )
+
+    return result
+  }
+
+  /**
+   * Orchestrator: For a teamSlug, get product -> component -> firing alert count
+   * @param teamSlug string
+   * @param alertsService alerts service instance
+   * @returns { [productName: string]: { [componentName: string]: number } }
+   */
+  async getTeamAlertSummary(
+    teamSlug: string,
+    alertsService: AlertsService,
+  ): Promise<{ [productName: string]: { [componentName: string]: number } }> {
+    logger.info(`[getTeamAlertSummary] Start for team ${teamSlug}`)
+
+    const products = await this.getProductsForTeam(teamSlug)
+    const productComponentMap = await this.getComponentsForProducts(products)
+
+    // Flatten all component names
+    const allComponentNames = Object.values(productComponentMap)
+      .flat()
+      .map(c => c.attributes.name)
+    logger.info(`[getTeamAlertSummary] Total components found: ${allComponentNames.length}`)
+
+    const alertCounts = await this.getFiringAlertCountsForComponents(allComponentNames, alertsService)
+
+    // Build the result using functional constructs
+    const result = Object.fromEntries(
+      Object.entries(productComponentMap).map(([productName, components]) => [
+        productName,
+        Object.fromEntries(
+          components.map(c => {
+            const componentName = c.attributes.name
+            return [componentName, alertCounts[componentName] || 0]
+          }),
+        ),
+      ]),
+    )
+
+    logger.info(`[getTeamAlertSummary] Done for team ${teamSlug}`)
+    return result
+  }
+
+  /**
+   * Utility: Get count of currently firing alerts for each component in a list
+   * @param componentNames Array of component names
+   * @param alertsService Alerts service instance (must have getAlerts())
+   * @returns Map of component name to count of firing alerts
+   */
+  async getFiringAlertCountsForComponents(
+    componentNames: string[],
+    alertsService: AlertsService,
+  ): Promise<{ [componentName: string]: number }> {
+    // This assumes alertsService.getAlerts() returns all alerts with .labels.application or .labels.component
+    try {
+      const allAlerts = await alertsService.getAlerts()
+      logger.info(`[getFiringAlertCountsForComponents] Total alerts fetched: ${allAlerts.length}`)
+
+      const nameSet = new Set(componentNames)
+      const countsMap = new Map<string, number>()
+
+      for (const alert of allAlerts) {
+        // Replace continue with conditional logic
+        if (alert.status?.state === 'active') {
+          const label = alert.labels?.application ?? alert.labels?.component
+          if (label && nameSet.has(label)) {
+            countsMap.set(label, (countsMap.get(label) || 0) + 1)
+          }
+        }
+      }
+
+      // Single pass for result + logging
+      return componentNames.reduce(
+        (acc, name) => {
+          acc[name] = countsMap.get(name) || 0
+          logger.info(`Component ${name}: ${acc[name]} firing alerts`)
```

Comment:
Is there any reason for this loop? Just log the list of names and return a joined list?

Reply:
Might be misinterpreting.